pkg/apis/core/validation/validation.go

```diff
@@ -2864,6 +2864,11 @@ func validateContainers(containers []core.Container, isInitContainers bool, volu
 			allErrs = append(allErrs, validateLifecycle(ctr.Lifecycle, idxPath.Child("lifecycle"))...)
 		}
 		allErrs = append(allErrs, validateProbe(ctr.LivenessProbe, idxPath.Child("livenessProbe"))...)
+		// Readiness-specific validation
+		if ctr.ReadinessProbe != nil && ctr.ReadinessProbe.TerminationGracePeriodSeconds != nil {
```

Comment:
should this validation depend on the feature gate? (being on or off)

Reply:
Hmm, it's possible for the feature flag to be off but for this to still be defined in the DB, and in any case it's disallowed, so I don't think we should gate it. WDYT?

Reply:
The gate would only be needed if we allowed this to be set at any point, but since we're starting without it, I think we're ok.  As long as there is no scenario with the gate on where someone could get a readinessprobe with this set, we don't have to worry about it.

Reply:
I expected to see a change in validateProbe that ensures the TerminationGracePeriodSeconds if set is non-negative.

Reply:
We are currently missing validation for the pod-level TerminationGracePeriodSeconds, so this is at least consistent...

See: https://github.com/kubernetes/kubernetes/pull/98866

That PR is going to miss code freeze, do we want to improve test coverage and fix it during test freeze?

Reply:
+1 on @derekwaynecarr 

But ok to address it after the code freeze.